File: src/algorithms/PlusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &d) {
        int len = d.size();
        d[len - 1]++;
        for (int i = len - 1; i > 0; i--) {
            if (d[i] >= 10) {
                d[i] %= 10;
                d[i - 1] ++;
            } else break;
        }
        if (d[0] >= 10) {
            d[0] %= 10;
            d.insert(d.begin(), 1);
        }
        return d;
    }
};

File: Algorithms/python/066-PlusOne/PlusOne.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """

        if len(digits) == 0:
            return digits

        s = len(digits) - 1
        digits[s] += 1
        t = 0

        for i in range(s, -1, -1):
            digits[i] += t
            t = digits[i] / 10
            if digits[i] < 10:
                break
            digits[i] -= 10
        if t > 0:
            return [t] + digits[:]
        return digits
